### ISSUE TITLE
error: ‘isnan’ was not declared in this scope

### ISSUE THREAD

**Issue by tomimaki** [scp-fs2open/fs2open.github.com#617]
After upgrade to Xubuntu 16.04 during compiling debug build I got this error:
```
In file included from ./asteroid/asteroid.h:16:0,
                 from asteroid/asteroid.cpp:12:
asteroid/asteroid.cpp: In function ‘object* asteroid_create(asteroid_field*, int, int)’:
./windows_stub/config.h:280:30: error: ‘isnan’ was not declared in this scope
 #define _isnan(f)     isnan(f)
                              ^
./globalincs/pstypes.h:263:9: note: in definition of macro ‘Assert’
   if (!(expr)) {\
         ^
./math/vecmat.h:19:27: note: in expansion of macro ‘_isnan’
 #define vm_is_vec_nan(v) (_isnan((v)->xyz.x) || _isnan((v)->xyz.y) || _isnan((v)->xyz.z))
                           ^
asteroid/asteroid.cpp:397:11: note: in expansion of macro ‘vm_is_vec_nan’
  Assert( !vm_is_vec_nan(&objp->phys_info.vel) ); 
           ^
./windows_stub/config.h:280:30: note: suggested alternative:
 #define _isnan(f)     isnan(f)
                              ^
./globalincs/pstypes.h:263:9: note: in definition of macro ‘Assert’
   if (!(expr)) {\
         ^
./math/vecmat.h:19:27: note: in expansion of macro ‘_isnan’
 #define vm_is_vec_nan(v) (_isnan((v)->xyz.x) || _isnan((v)->xyz.y) || _isnan((v)->xyz.z))
                           ^
asteroid/asteroid.cpp:397:11: note: in expansion of macro ‘vm_is_vec_nan’
  Assert( !vm_is_vec_nan(&objp->phys_info.vel) ); 
           ^
In file included from /usr/include/c++/5/random:38:0,
                 from /usr/include/c++/5/bits/stl_algo.h:66,
                 from /usr/include/c++/5/algorithm:62,
                 from asteroid/asteroid.cpp:45:
/usr/include/c++/5/cmath:641:5: note:   ‘std::isnan’
     isnan(_Tp __x)
     ^
  CXX      cfile/cfile.o
```
Some searching suggests it's because change in glibc.
https://sourceware.org/bugzilla/show_bug.cgi?id=17441
I added
`#include <cmath>` and changed `#define _isnan(f)     isnan(f)` to `#define _isnan(f)     std::isnan(f)`
and it works.
But I don't if this is correct solution and how it will affect compiling with older glibc.

**Comment by niffiwan** [comment]
From reading this: https://sourceware.org/bugzilla/show_bug.cgi?id=19439

It seems that your suggestion is the correct solution, seems like valid/proper C++, just needs testing with older versions of GCC, Clang & MSVC

**Comment by asarium** [comment]
@tomimaki: Could you submit your change as a PR? The CI servers will let us know if it works with older version of GCC and MSVC.

**Comment by asarium** [comment]
This has been fixed by #621.
